HW2/P3/mandelbrot.pyx

```diff
@@ -0,0 +1,66 @@
+cimport numpy as np
+cimport cython
+import numpy
+cimport AVX
+from cython.parallel import parallel, prange
+from libc.stdlib cimport abort, malloc, free
+
+cdef np.float64_t magnitude_squared(np.complex64_t z) nogil:
+    return z.real * z.real + z.imag * z.imag
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cpdef mandelbrot(np.complex64_t [:, :] in_coords,
+                 np.uint32_t [:, :] out_counts,
+                 int max_iterations=511):
+    cdef:
+       int i, j, k, h #iter
+       np.complex64_t c, z
+       AVX.float8 vcreal,vcimag,vzreal,vzimag,tmpreal
+       AVX.float8 threshold,iter_flag,iter,cmp,max_iter,mag_sq
+       float out[8]
+       # To declare AVX.float8 variables, use:
+       # cdef:
+       #     AVX.float8 v1, v2, v3
+       #
+       # And then, for example, to multiply them
+       #     v3 = AVX.mul(v1, v2)
+       #
+       # You may find the numpy.real() and numpy.imag() fuctions helpful.
+
+    assert in_coords.shape[1] % 8 == 0, "Input array must have 8N columns"
+    assert in_coords.shape[0] == out_counts.shape[0], "Input and output arrays must be the same size"
+    assert in_coords.shape[1] == out_counts.shape[1],  "Input and output arrays must be the same size"
+    
+    with nogil, parallel(num_threads=2):
+      # malloc space for buffer
+      local_buf=<float *> malloc(sizeof(float)*8)
```

Comment:
Do you have to malloc here? It might be easier to just use a numpy array of the desired size where memory will be handled for you